### PR TITLE
Use CMake to generate cmakein file

### PR DESCRIPTION
**Pull request by donny-dont**
When building on a Windows host `sed` is not available. This patch uses CMake commands directly to create the zconf.h.cmakein file.

### CODE REVIEW

**Comment by donny-dont** [comment]
@Dead2 @mtl1979 you guys should be able to setup https://www.appveyor.com/ if you want to do Windows builds after this patch lands.

**Comment by mtl1979** [comment]
@donny-dont I actually run Windows 10, so I can do Windows build even locally.

**Comment by donny-dont** [comment]
@mtl1979 I was referring to having a CI. I don't believe travis supports windows at all and I have used appveyor for doing windows builds. As this patch removes the requirement of having `sed` then a CI for windows builds would be possible.

File: CMakeLists.txt [hunk 1]
```diff
@@ -454,11 +454,16 @@ message(STATUS "Architecture-specific source files: ${ZLIB_ARCH_SRCS}")
 #============================================================================
 
 macro(generate_cmakein input output)
-    execute_process(COMMAND sed "/#define ZCONF_H/ a\\\n#cmakedefine Z_HAVE_UNISTD_H\\\n#cmakedefine Z_HAVE_STDARG_H\n"
-                    INPUT_FILE ${input}
-                    OUTPUT_FILE ${output}
-)
-
+    file(REMOVE ${output})
+    file(STRINGS ${input} _lines)
```
Comment (donny-dont) [inline CMakeLists.txt:458]: `file(STRINGS)` is available in CMake 2.8.4

https://cmake.org/cmake/help/v2.8.4/cmake.html#command:file

File: CMakeLists.txt [hunk 2]
```diff
@@ -454,11 +454,16 @@ message(STATUS "Architecture-specific source files: ${ZLIB_ARCH_SRCS}")
...
+    foreach(_line IN LISTS _lines)
```
Comment (donny-dont) [inline CMakeLists.txt:459]: `foreach(IN)` is available in CMake 2.8.4.

https://cmake.org/cmake/help/v2.8.4/cmake.html#command:foreach

**Review by donny-dont — COMMENTED** [review]
Responding to @Dead2 's comments in IRC.

This setup would work in the minimum CMake required, 2.8.4. It produces identical output with the current `sed` command as it just appends the two `#cmakedefin` lines.

**Comment by mtl1979** [comment]
@donny-dont : @Dead2 handles the updating of CI configurations, but Windows builds are my responsibility. Even if we add CI for Windows builds, it's still easier to diagnose possible issues locally.

**Pull request merged**